doc/source/text.rst

```diff
@@ -205,6 +217,52 @@ and optional groups like
 
 can also be used.
 
+When ``str.extract`` method is used on an ``Index``, it returns a
+``DataFrame`` with as many columns as capture groups.
```

Comment:
Do I need to clarify the docs? I thought it was clear enough from the first examples without named groups, followed by the discussion of named groups.